app/src/main/kotlin/org/fossify/notes/activities/MainActivity.kt

```diff
@@ -1557,4 +1564,24 @@ class MainActivity : SimpleActivity() {
             updateWidgets()
         }
     }
+
+    private fun checkReadOnlyState() {
+        getCurrentFragment()?.apply {
+            if (this is TextFragment) {
+                (this as TextFragment).getNotesView().isEnabled = !mCurrentNote.isReadOnly
+            }
+        }
+    }
+
+    private fun toggleReadOnly() {
+        mCurrentNote.isReadOnly = !mCurrentNote.isReadOnly
+        NotesHelper(this).insertOrUpdateNote(mCurrentNote) {
+            refreshMenuItems()
+            getCurrentFragment()?.apply {
+                if (this is TextFragment) {
+                    (this as TextFragment).getNotesView().isEnabled = !mCurrentNote.isReadOnly
+                }
+            }
```

Comment:
You could replace this duplicated code with a call to `checkReadOnlyState()`:

```suggestion
            checkReadOnlyState()
```